integration/airflow/openlineage/airflow/extractors/sql_extractor.py

```diff
@@ -103,18 +103,44 @@ def extract(self) -> TaskMetadata:
         # as the current connection. We need to also fetch the schema for the
         # input tables to format the dataset name as:
         # {schema_name}.{table_name}
+        in_tables = [t for t in sql_meta.in_tables if t.table_type == TableType.Table]
+        out_tables = [t for t in sql_meta.out_tables if t.table_type == TableType.Table]
         inputs, outputs = get_table_schemas(
             self.hook,
             source,
             database,
-            self._information_schema_query(sql_meta.in_tables)
-            if sql_meta.in_tables
-            else None,
-            self._information_schema_query(sql_meta.out_tables)
-            if sql_meta.out_tables
-            else None,
+            self._information_schema_query(in_tables) if in_tables else None,
+            self._information_schema_query(out_tables) if out_tables else None,
         )
 
+        # (4) Map external stage tables to datasets
+        # (no need to query information schema)
+        for in_table in sql_meta.in_tables:
+            if in_table.table_type == TableType.ExternalLocation:
+                inputs.append(Dataset.from_table(
+                    source=source,
+                    table_name=in_table.name
+                ))
+            elif in_table.table_type == TableType.Stage:
+                inputs.append(Dataset.from_table(
+                    source=source,
+                    table_name=in_table.name,
+                    database_name=database,
+                ))
+        for out_table in sql_meta.out_tables:
+            if out_table.table_type == TableType.ExternalLocation:
+                outputs.append(Dataset.from_table(
+                    source=source,
+                    table_name=out_table.name
+                ))
+            elif out_table.table_type == TableType.Stage:
+                outputs.append(Dataset.from_table(
+                    source=source,
+                    table_name=out_table.name,
+                    schema_name=out_table.schema,
+                    database_name=database,
+                ))
```

Comment:
We might wrap it up a little bit? Lots of repeated code going on here.
e.g. a method to create Dataset from sqlparser's table.